### DISCUSSION TITLE
I cannot Importing Assets with Chinese character

### DISCUSSION THREAD

**Discussion by cnah66** [snipe/snipe-it#10465] [Q&A]
when I import a .csv file with Chinese character in it. it will be failed.
what can I do ?
thank you !

**Comment by cnah66** [comment]
my sever environment
windows11
wamp 3.2.6 x64
mysql :5.7.36
php:7.4.26

**Reply by yosiasz** [reply]
what are you using for your database server: mysql? what is your database collation?

**Reply by cnah66** [reply]
mysql :5.7.36  ， database collation is  utf8mb4_unicode_ci

**Comment by yosiasz** [comment]
https://github.com/snipe/snipe-it/issues/2921

**Reply by cnah66** [reply]
[sample-assets - Chinese.csv](https://github.com/snipe/snipe-it/files/7783334/sample-assets.-.Chinese.csv)
I read this. but it seems have no solution yet.
now I save my CSV file as CSV UTF-8, it become better now , but still have a problem.
Chinese name & Email  these two field will not break. they are together.  this problem makes the below fields are all wrong.

![chinese CSV UTF-8](https://user-images.githubusercontent.com/95849938/147538285-c3bae65e-2ac1-4b67-aee9-05c7d16db4a2.png)
![name with email not break](https://user-images.githubusercontent.com/95849938/147538291-802152ae-d031-4963-b181-152e669c4043.png)

the CSV file with Chinese in it :
[sample-assets - Chinese.csv](https://github.com/snipe/snipe-it/files/7783338/sample-assets.-.Chinese.csv)

the CSV file with Chinese in it looks like this:
![image](https://user-images.githubusercontent.com/95849938/147538569-933f0e2a-1251-4e79-ae5f-a1688367cb13.png)

**Comment by cnah66** [comment]
&nbsp;
#2921&nbsp; haven't solve my problem. Is this still a bug?
Thank you for your help.

**Comment by King-GO** [comment]
Yes, i've been experienced the same issue when imported the UTF-8 CSV file with the field with Chinese character. (I'd tested a couple of times and then gave up)

here are my solutions, 
1, translate your Chinese to English before imported (i know this is a big task, but it's save your time when compare to troubleshooting the outstanding issues)
2, try to switch the columns, shift the Chinese field to the end or the start of the first column.

**Reply by cnah66** [reply]
2, try to switch the columns, shift the Chinese field to the end or the start of the first column.

my problem is:
the column of Chinese full name at the first column will be imported "Chinese full name, the second column date" together as the full name column.

**Reply by cnah66** [reply]
兄弟，你有微信吗？想跟聊聊这个软件，我的微信号：cnah66

**Comment by King-GO** [comment]
I know that sometime two fields will combine together, just like your case. 
what about exchange A column (Full name) to R column (Supplier) ?

**Reply by cnah66** [reply]
I don't know why.. it sometimes works.

**Comment by yosiasz** [comment]
please provide sample csv file so we can test it on our side to see if it is an issue we can reproduce. images are not going to help

**Reply by King-GO** [reply]
cnah66 is already provided [https://github.com/snipe/snipe-it/files/7783334/sample-assets.-.Chinese.csv](url)

**Reply by yosiasz** [reply]
nope, cannot access that file

**Reply by yosiasz** [reply]
Was able to import with no issues via snipeit UI. 

![image](https://user-images.githubusercontent.com/900951/147979886-81603056-3e71-4003-896f-11391178219e.png)

**Reply by yosiasz** [reply]
SELECT @@character_set_database, @@collation_database, @@collation_server

'utf8mb4', 'utf8mb4_0900_ai_ci', 'utf8mb4_0900_ai_ci'

**Reply by King-GO** [reply]
How to SELECT @@character_set_database, @@collation_database, @@collation_server via phpmyadmin?
when create the database (xampp), the database collation has selected **utf8mb4_general_ci** already.

**Reply by cnah66** [reply]
> Was able to import with no issues via snipeit UI.
> 
> ![image](https://user-images.githubusercontent.com/900951/147979886-81603056-3e71-4003-896f-11391178219e.png)

Thank you for your help. the Full name column  is the first column ?   in my side the first column  and the second column will be combine together.
Maybe because I use a Chinese version windows?
I use King-GO's solution ,  put the Full name column to the last , now is OK.

**Comment by cnah66** [comment]
problem csv file:
[new test - 副本.csv](https://github.com/snipe/snipe-it/files/7806033/new.test.-.csv)

also have a problem :(

![微信截图_20220104152404](https://user-images.githubusercontent.com/95849938/148023696-89b5f744-56aa-4f7a-a76d-89a5a71e3903.png)

**Reply by yosiasz** [reply]
works just fine on snipeit v5.3.6 - build 6567 

![image](https://user-images.githubusercontent.com/900951/148421665-f37cf896-cfb9-4b96-9435-0d42baa12f3f.png)

**Reply by yosiasz** [reply]
``
Email,Username,item Name,Category,Model name,Manufacturer,Model Number,Serial number,Asset Tag,Location,Notes,Purchase Date,Purchase Cost,Company,Status,Warranty,Supplier,Full Name
,,办公电脑 ,笔记本,DELL-5410,,,,HF012038,班子成员,,,,肥东国轩新材料有限公司,Undeployable,,总部,彭立煌
``
